sm/plugin_sm/src/plugin.rs

```diff
@@ -49,13 +49,30 @@ pub trait Plugin {
 pub struct ExternalPluginCommand {
     pub name: SoftwareType,
     pub path: PathBuf,
+    pub sudo: Option<PathBuf>,
 }
 
 impl ExternalPluginCommand {
     pub fn new(name: impl Into<SoftwareType>, path: impl Into<PathBuf>) -> ExternalPluginCommand {
-        ExternalPluginCommand {
-            name: name.into(),
-            path: path.into(),
+        let sw_type = &name.into().clone() as &str;
+        match sw_type {
+            "test" => {
```

Comment:
Don't hack the main code base for testing purpose.

* Remove this `if test` check.
* Either add a third parameter to get the path to `sudo`,
* or directly build an `ExternalPluginCommand` struct from the test without calling `ExternalPluginCommand::new()`.

Reply:
I fixed it by using the last option, creating an instance of the ExternalPluginCommand for the test without using the new function.